lrtable/src/lib/statetable.rs

```diff
@@ -303,33 +308,25 @@ where
     }
 
     fn decode(bits: u64) -> Action<StorageT> {
-        let action = bits & 0b11;
-        let reference = (bits >> 2) as u32;
+        let action = (bits & 0b11) as u8;
+        let val = (bits >> 2) as u32;
```

Comment:
Does `let val: u64 = bits >> 2` not work? Similarly for the line above.

Reply:
No, because `StIdx` can't convert from `u64` unless I add a `from_u64` to it.

Reply:
OK then `let val: u32 = ...`.